meson.build

```diff
@@ -0,0 +1,246 @@
+        '-Wstrict-prototypes',
+        '-Wpointer-arith',
+        '-Wredundant-decls',
+        '-Wno-missing-field-initializers'
+    ]
+    add_project_arguments(cc.get_supported_arguments(cflags), language: 'c')
+endif
+
+test_args = []
+
+if host_machine.system() == 'linux'
+    test_args += '-D_GNU_SOURCE'
+    add_project_arguments('-D_GNU_SOURCE', language: 'c')
+elif host_machine.system() == 'darwin'
+    test_args += '-D_DARWIN_C_SOURCE'
+    add_project_arguments('-D_DARWIN_C_SOURCE', language: 'c')
+elif cc.get_id() != 'msvc'
+    test_args += '-D_POSIX_C_SOURCE=200809L'
+    add_project_arguments('-D_POSIX_C_SOURCE=200809L', language: 'c')
+endif
+
+# Configuration
+
+if get_option('large-tiles')
+    conf.set('CONFIG_LARGE_TILES', 1)
+endif
+
+conf.set('PACKAGE_NAME', 'libass')
+conf.set('PACKAGE_VERSION', meson.project_version())
+
+conf.set('CONFIG_SOURCEVERSION', '"meson, on or after: ' + meson.project_version() + '"')
+
+str_check_functions = [
+    'strdup',
+    'strndup'
+]
+
+foreach name: str_check_functions
+    if cc.has_function(name) and cc.has_header_symbol('string.h', name, args: test_args)
+        conf.set('HAVE_@0@'.format(name.to_upper()), 1)
+    endif
+endforeach
+
+# Dependencies
+
+m_dep = cc.find_library('m', required: false)
+if m_dep.found()
+    deps += m_dep
+endif
+
+iconv_dep = cc.find_library('iconv', required: false)
+if iconv_dep.found()
+    deps += iconv_dep
+endif
+if iconv_dep.found() or cc.has_function('iconv_open')
+    conf.set('CONFIG_ICONV', 1)
+endif
+
+freetype_dep = dependency('freetype2',version: '>= 2.10.92',
+                          default_options: ['default_library=' + get_option('default_library'), 'harfbuzz=disabled']) # workaround for https://github.com/mesonbuild/meson/issues/8047
+if freetype_dep.found()
+    deps += freetype_dep
+    conf.set('CONFIG_FREETYPE', 1)
+endif
+
+if host_machine.system() == 'windows' and get_option('default_library') == 'static'
+    add_project_arguments('-DFRIBIDI_LIB_STATIC', language: 'c')
+endif
+fribidi_dep = dependency('fribidi', version: '>= 0.19.0',
+                         default_options: ['docs=false', 'tests=false'])
+if fribidi_dep.found()
+    deps += fribidi_dep
+    conf.set('CONFIG_FRIBIDI', 1)
+endif
+
+harfbuzz_options = ['tests=disabled', 'cairo=disabled', 'gobject=disabled']
+if host_machine.system() == 'windows' or host_machine.system() == 'darwin'
+    harfbuzz_options += ['glib=disabled', 'freetype=disabled']
+endif
+harfbuzz_dep = dependency('harfbuzz', version: '>= 1.2.3', required: get_option('harfbuzz'),
+                          fallback: ['harfbuzz', 'libharfbuzz_dep'],
+                          default_options: harfbuzz_options)
+if harfbuzz_dep.found()
+    deps += harfbuzz_dep
+    conf.set('CONFIG_HARFBUZZ', 1)
+endif
+
+req_fontconfig = get_option('fontconfig')
+
+if host_machine.system() != 'windows' and host_machine.system() != 'darwin'
+    req_fontconfig = get_option('fontconfig').disabled()
```

Comment:
What is this doing?

If you're on Unix-and-not-macOS, and you configure with -Dfontconfig=disabled, then req_fontconfig is true and the dependency lookup is required, but if you configure with =enabled or =auto, then it is not required?

Reply:
I maybe just missed a not here? IIRC the intention is that unless you specifically enable it, fontconfig doesn't get brought in on Windows and MacOS, since we prefer DirectWrite and CoreText respectively. I'll look at this logic when I get around to updating this PR.

Reply:
A missing `not` will simply convert auto() and enabled() to boolean "true" and disabled() to boolean "false", when what I think you wanted to do is make auto() act like disabled() and completely skip it even if found.

> `disable_auto_if(value)` (since 0.59.0): returns the feature, with `'auto'` converted to `'disabled'` if value is true.

does what you want, I think.

Reply:
> unless you specifically enable it, fontconfig doesn't get brought in on Windows and MacOS, since we prefer DirectWrite and CoreText respectively.

This doesn't match the existing autoconf logic fwiw; also the fontconfig flag apparently defaults to `'disabled'` if not set explicitly? *(Which also doesn't match any other fontprovider in neither autoconf nor this Meson port.)*

Reply:
Ugh, I see. Is the existing autoconf behavior assumed to be correct, or should we define separately what the intended behavior is and update both build systems? I have to admit I don't particularly enjoy repeatedly having to go fix up this PR to match reworked autoconf logic, so it would be great if we could nail this down. cc: @astiob

Reply:
Fontconfig has always been added if found unless specifically disabled like all other optional dependencies of the library. Nothing changed in this regard and I don't see a reason why this should be changed.

Reply:
Yup, what Oneric said. The Autotools setup adds Fontconfig if found and not explicitly disabled regardless of platform, and it has always worked like this. I do appreciate the sentiment, but this one time we didn’t change anything :-)

Reply:
Ok great, will fix.